models/issue_reaction.go

```diff
@@ -77,6 +100,10 @@ type ReactionOptions struct {
 
 // CreateReaction creates reaction for issue or comment.
 func CreateReaction(opts *ReactionOptions) (reaction *Reaction, err error) {
+	if !util.IsStringInSlice(opts.Type, setting.UI.Reactions) {
```

Comment:
This could be very inefficient.

I suspect we should be converting `setting.UI.Reactions` to a `map[string]bool{}` eg within the appropriate setting file: 

```go
ReactionsMap = make(map[string]bool{})
for _, reaction := range Reactions {
    ReactionsMap[reaction] = true
}
```

Then this becomes:

```go
if !setting.UI.ReactionsMap[opts.Type] {
...
```

Reply:
do 70c41764b8892144541d9978017bf62f87cd9508 resolve this?

Reply:
@zeripath do 567af8415725da7a22261d4821bd90614abd1c4e resolve this?

Reply:
yup!